packages/public/common-ui-web/src/components/Slider/useSlider.ts

```diff
@@ -0,0 +1,118 @@
+import { useState, RefObject, useLayoutEffect } from 'react';
+
+function getFirstThumbPosition(min: number, max: number, value: number): number {
+  if (value > max) {
+    return 100;
+  }
+  if (value < min) {
+    return 0;
+  }
+  return ((value - min) / (max - min)) * 100;
+}
+
+function getThumbPosition(
+  event: MouseEvent | TouchEvent,
+  sliderRef: RefObject<HTMLDivElement>,
+  step: number,
+  min: number,
+  max: number,
+): number {
+  if (!sliderRef.current) {
+    return 0;
+  }
+  const sliderElement = sliderRef.current;
+  const sliderRect = sliderElement.getBoundingClientRect();
+  const offsetX = Math.max(
+    0,
+    Math.min(
+      sliderRect.width,
+      (event instanceof MouseEvent ? event.clientX : event.touches[0].clientX) - sliderRect.left,
+    ),
+  );
+  const positionPercentage = (offsetX / sliderRect.width) * 100;
+  const stepPercentage = 100 / ((max - min) / step);
+  return Math.round(positionPercentage / stepPercentage) * stepPercentage;
+}
+
+function getNewSliderValue(
+  roundedPercentage: number,
+  max: number,
+  min: number,
+  step: number,
+): number {
+  let multiplier = 1;
+  if (!Number.isInteger(step)) {
+    const nbDigitAfterDot = step.toString().split('.')[1].length;
+    multiplier = 10 ** nbDigitAfterDot;
+  }
+  return Math.round(((max - min) * (roundedPercentage / 100) + min) * multiplier) / multiplier;
+}
+
+export interface useSliderParams {
```

Comment:
Capital letter for this interface : `UseSliderParams`

Reply:
eslint naming convention added to get rid of this error